esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -2093,6 +2093,22 @@ extern TPWRAPPER_API void initContainerRoxieTargets(MapStringToMyClass<ISmartSoc
     }
 }
 
+extern TPWRAPPER_API void queryQueriesOnlyRoxieTargetNames(StringArray& names)
+{
+    CConstWUClusterInfoArray clusters;
+    getEnvironmentClusterInfo(clusters);
+    ForEachItemIn(i, clusters)
+    {
+        IConstWUClusterInfo& clusterInfo = clusters.item(i);
+        if (clusterInfo.getPlatform() == RoxieCluster)
+        {
```

Comment:
If as mentioned above you change the name to something like "getRoxieTargetsSupportingPublishedQueries()" you may want to check clusterInfo.canPublishQueries() before adding to list?